### PR TITLE
[Merged by Bors] - conservative cache: reset accounts that only have infeasible txs in db

### LINKED ISSUE #3663: conservative cache: insufficient balance is not re-evaluated after layers are applied
## Description

@brusherru / @dshulyak reported that tx that was first not accepted by mempool due to insufficient funds are not reconsidered after layers are applied (and received rewards / incoming funds), and somehow prevent submitting other transactions even if nonce was incremented.

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation

Closes #3663

## Changes
- added two tests that failed with the old code
- after a layer is applied, also reset accounts that ONLY have infeasible txs in db, as they may also receive new funds/rewards in the most recent applied layer

### CODE REVIEW

File: txs/cache.go [hunk 1]
```diff
@@ -727,13 +732,16 @@ func (c *cache) ApplyLayer(
 		}
 		acctResetDuration.Observe(float64(time.Since(t1)))
 	}
-	for principal := range skippedByPrincipal {
-		if _, ok := byPrincipal[principal]; ok {
+
+	for principal, cash := range c.pending {
```
Comment (dshulyak) [inline txs/cache.go]: what is cash? 

i looked up in the code, should it be called account? i can't make any connection to cash

**Review by dshulyak — APPROVED** [review]
is there any eviction policy for infeasible transactions from disk and accounts that have only infeasible transactions from that pending map? 

 this must be a problem, if there is no eviction and no limit on the number of accounts

**Comment by countvonzero** [comment]
> is there any eviction policy for infeasible transactions from disk and accounts that have only infeasible transactions from that pending map?
> 
> this must be a problem, if there is no eviction and no limit on the number of accounts

i thought i had previously filed an issue for this but i cannot find it now. in any case, i filed https://github.com/spacemeshos/go-spacemesh/issues/3668 with more detailed description and implementation proposals. please review. thanks

File: txs/cache.go (hunk 1 above)
Comment (countvonzero) [reply txs/cache.go]: don't want to call it cache (clash with the struct name) so use it as it sounds the same. anyhow, renamed to `accCache`

**Comment by countvonzero** [comment]
bors merge

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**